includes/class-search.php

```diff
@@ -0,0 +1,154 @@
+			return $query;
+		}
+
+		$search_term = $query->get( 's' );
+		if ( empty( $search_term ) ) {
+			return $query;
+		}
+
+		// Check if search term is a URL.
+		if ( ! \wp_http_validate_url( $search_term ) ) {
+			return $query;
+		}
+
+		// Try to import ActivityPub object.
+		$imported = self::try_import_activitypub_object( $search_term );
+
+		if ( $imported ) {
+			// Ensure the imported comment is approved/published.
+			\wp_set_comment_status( $imported, 'approve' );
+			$comment_link   = \get_comment_link( $imported );
+			$validated_link = \wp_validate_redirect( $comment_link, \home_url() );
+			if ( $validated_link ) {
+				\wp_safe_redirect( $validated_link );
+				exit;
+			}
+		}
+
+		// Fall back to classic search if import failed or no redirect.
+		return $query;
+	}
+
+	/**
+	 * Handle admin comment search to check for URLs and ActivityPub objects.
+	 * Runs on admin_init to avoid infinite loops.
+	 */
+	public static function enhance_admin_comment_search() {
+		// Check for a valid user session.
+		if ( ! is_user_logged_in() ) {
+			return;
+		}
+
+		// Only enhance admin comment searches.
+		if ( ! \is_admin() ) {
+			return;
+		}
+
+		// Check if we're on the edit-comments.php page with a search.
+		global $pagenow;
+		if ( 'edit-comments.php' !== $pagenow ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$search_term = isset( $_GET['s'] ) ? \sanitize_text_field( \wp_unslash( $_GET['s'] ) ) : '';
+		if ( empty( $search_term ) ) {
+			return;
+		}
+
+		// Check if search term is a URL.
+		if ( ! \wp_http_validate_url( $search_term ) ) {
+			return;
+		}
+
+		// Try to import ActivityPub object.
+		$imported = self::try_import_activitypub_object( $search_term );
+
+		if ( $imported ) {
+			// Ensure the imported comment is approved/published.
+			\wp_set_comment_status( $imported, 'approve' );
+			$comment_link   = \get_comment_link( $imported );
+			$validated_link = \wp_validate_redirect( $comment_link, \home_url() );
+			if ( $validated_link ) {
+				\wp_safe_redirect( $validated_link );
+				exit;
+			}
+		}
+	}
+
+	/**
+	 * Try to import an ActivityPub reply from a URL.
+	 *
+	 * @param string $url The URL to check and import.
+	 *
+	 * @return int|false The imported comment ID or false on failure.
+	 */
+	private static function try_import_activitypub_object( $url ) {
+		// Check if it's already imported.
+		$existing = Comment::url_to_commentid( $url );
+		if ( $existing ) {
+			return $existing;
+		}
+
+		// Try to fetch as ActivityPub object.
+		$object = Http::get_remote_object( $url );
+		if ( \is_wp_error( $object ) ) {
+			return false;
+		}
+
+		// Check if it's a reply (has inReplyTo).
+		if ( empty( $object['inReplyTo'] ) ) {
+			return false;
+		}
+
+		$activity = array(
+			'type'   => 'Create',
+			'actor'  => $object['attributedTo'],
+			'object' => $object,
+		);
+
+		// Import the reply as a comment.
+		return Interactions::add_comment( $activity );
```

Comment:
I noticed that the comment date/time is the time when the comment gets imported, not when it was created originally. I assume this would need updating down the stack.

Reply:
@obenland https://github.com/Automattic/wordpress-activitypub/pull/2038